### PR TITLE
Fixes #3605 Replace deprecated jQuery methods

### LINKED ISSUE #3605: jquery 3.5.1 compatibility and WP 5.7
**- Made sure you’re on the latest version**
Yes

**Describe the bug**
The Wordpress jquery migration helper
[https://wordpress.org/plugins/enable-jquery-migrate-helper/](https://wordpress.org/plugins/enable-jquery-migrate-helper/)
report the following jquery 3.5.1 errors for
for: /wp-rocket/assets/js/wpr-modal.js

On March, 09 WP 5.7 needs jquery 3.5.1 compatibility.

wp-rocket/assets/js/wpr-modal.js: jQuery.fn.click() event shorthand is deprecated
wp-rocket/assets/js/wpr-modal.js: jQuery.fn.bind() is deprecated
wp-rocket/assets/js/wpr-modal.js: jQuery.fn.change() event shorthand is deprecated
wp-rocket/assets/js/wpr-modal.js: jQuery.fn.keyup() event shorthand is deprecated

**Solutions**
Solution: Instead of .click(fn) use .on("click", fn). Instead of .click() use .trigger("click")
Solution: Instead of .bind(fn) use .on(fn)
Solution: Instead of .change(fn) use .on("change", fn). Instead of .change() use .trigger("change")
Solution: Instead of .keyup(fn) use .on("keyup", fn)

**See also**
[https://github.com/jquery/jquery-migrate/blob/master/warnings.md](https://github.com/jquery/jquery-migrate/blob/master/warnings.md)

**Additional context**
.click() shorthand is deprecated: https://api.jquery.com/click/#click-handler
.bind() is deprecated: https://api.jquery.com/bind/
.change() is deprecated: https://api.jquery.com/change/#change-handler
.keyup() id deprecated: https://api.jquery.com/keyup/#keyup-handler

**Screenshot**
from wpr-modal.js
coding

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

### PR DESCRIPTION
**Pull request by remyperona**
## Description

Replace the deprecated jQuery methods by the corresponding ones for the current jQuery version. This is to make sure we are compatible with WordPress once jQuery migrate is removed from core (currently planned for 5.8)

Fixes #3605

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Is the solution different from the one proposed during the grooming?

No

## How Has This Been Tested?

- [x] Tested with `SCRIPT_DEBUG` enabled, and looking at the browser console to see warnings displayed on the `plugins.php` page and WP Rocket settings page

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by iCaspar — APPROVED** [review]

**Review by crystinutzaa — APPROVED** [review]

File: assets/js/wpr-modal.js [hunk 1]
```diff
@@ -62,50 +62,49 @@ function ModalWpr(aElem) {
...
     // Close
-    this.closer.click(function() {
+    this.closer.on( 'click', function() {
         refThis.close();
         return false;
     });
 
-    aElem.bind('keyup', function(){
-        if(event.keyCode == 27){ // ECHAP
+    aElem.bind('keyup', function(e){
```
Comment (engahmeds3ed) [inline assets/js/wpr-modal.js]: this old code isn't working at all, please try the following one is working.
```
$(document).on('keyup', function(e){
        if( aElem.length > 0 && 27 === e.keyCode ){ // ECHAP
            refThis.close();
            return false;
        }
    });
```

File: assets/js/wpr-modal.js [hunk 2]
```diff
@@ -62,50 +62,49 @@ function ModalWpr(aElem) {
...
-    this.radio.change(function(){
+    this.radio.on( 'change', function() {
         refThis.change($(this));
     });
 
     // Write text
-    this.textFields.keyup(function() {
+    this.textFields.on( "keyup", function() {
         refThis.hiddenDetails.val($(this).val());
         if(refThis.hiddenDetails.val() != ''){
```
Comment (engahmeds3ed) [inline assets/js/wpr-modal.js]:
```suggestion
        if( '' !== refThis.hiddenDetails.val() ){
```

**Review by engahmeds3ed — CHANGES_REQUESTED** [review]

**Review by engahmeds3ed — APPROVED** [review]

**Review by piotrbak — CHANGES_REQUESTED** [review]
@Tabrisrp When refreshing the license info we're still getting deprecation warning in the `wpr-admin.js`:
`JQMIGRATE: jQuery.fn.blur() event shorthand is deprecated`

**Comment by remyperona** [comment]
@piotrbak Thank you I missed this one, it's updated!

**Comment by piotrbak** [comment]
@Tabrisrp I scanned our files for the deprecated functions from the:
https://github.com/jquery/jquery-migrate/blob/master/warnings.md

I'm not sure whether they're JS or jQuery though, please examine them:
- `micromodal.min.js` `.bind()`
- `wpr-admin-common.js` `.ready()`
- `wpr-admin.js` `.ready()` `.easing()` `.type` `.isArray()` `.css()`
- `wpr-modal.js` `.ready()` `.css()`

If they're not related, then we're ready to merge it.

**Comment by remyperona** [comment]
@piotrbak Had a second look, everything is fine.

**Review by piotrbak — APPROVED** [review]

**Pull request merged**